src/Microsoft.ComponentDetection.Common/CommandLineInvocationService.cs

```diff
@@ -149,14 +176,29 @@ private static Task<CommandLineExecutionResult> RunProcessAsync(string fileName,
         process.Exited += (sender, args) =>
         {
             Task.WaitAll(t1, t2);
-            tcs.SetResult(new CommandLineExecutionResult { ExitCode = process.ExitCode, StdErr = errorText, StdOut = stdOutText });
+            tcs.TrySetResult(new CommandLineExecutionResult { ExitCode = process.ExitCode, StdErr = errorText, StdOut = stdOutText });
             process.Dispose();
         };
 
         process.Start();
         t1.Start();
         t2.Start();
 
+        cancellationToken.Register(() =>
```

Comment:
Does this need to be set before starting the process?

Reply:
No its not required. I did it this way so we would start the process, then it would immediately exit once we hit this registration and trigger the process.Exited event which disposes the process. Not sure if that same flow will happen if we trigger the process kill before its ever started

Reply:
tested this out, and indeed I'm not seeing disposal of the process afterwards if we cancel before the process is started